### PR TITLE
Ticket 22 : Add product router

### PR DESCRIPTION
**Pull request by maximFeldman1**
add react router dom + products url

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,17 @@
 import React from "react";
 import "./App.css";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
+import Products from "./Pages/ProductsPage";
```
Comment (amitmarko) [inline src/App.tsx]: Please add lazy and suspense

File: src/Pages/ProductsPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (amitmarko) [inline src/Pages/ProductsPage/index.tsx]: You names the folder `ProductsPage` but you are in `Pages` folder so it's redundant,
please change folder name to just `Products`

File: src/App.tsx [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
 
 function App() {
-  return <div className="App">test</div>;
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route path="/products" element={<Products />} />
```
Comment (amitmarko) [inline src/App.tsx]: Create URL constants file 
```suggestion
        <Route path={URL.PRODUCTS} element={<Products />} />
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
+        <Route path="/*" element={<Navigate to="/products" />} />
```
Comment (amitmarko) [inline src/App.tsx]:
```suggestion
        <Route path="/*" element={<Navigate to={URL.PRODUCTS} />} />
```

File: src/App.tsx (hunk 1 above)
Comment (amitmarko) [reply src/App.tsx:4]: Fixed

File: src/Pages/ProductsPage/index.tsx (hunk 2 above)
Comment (maximFeldman1) [reply src/Pages/ProductsPage/index.tsx]: fixed

File: src/App.tsx (hunk 1 above)
Comment (maximFeldman1) [reply src/App.tsx]: Fixed

File: src/App.tsx (hunk 4 above)
Comment (maximFeldman1) [reply src/App.tsx]: Fixed

File: src/Components/App/AppRouter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { URL } from "../../../shared";
+
+const Products = lazy(() => import("../../../Pages/Products"));
+
+const AppRouter = () => (
+  <Suspense fallback={<div>Loading...</div>}>
+    <Routes>
+      <Route path="/products" element={<Products />} />
+      <Route
+        path={URL.getProducts}
```
Comment (amitmarko) [inline src/Components/App/AppRouter/index.tsx]:
```suggestion
        path={URL.PRODUCTS}
```

File: src/Components/App/AppRouter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        element={<Navigate to={URL.getProducts} />}
```
Comment (amitmarko) [inline src/Components/App/AppRouter/index.tsx]:
```suggestion
        element={<Navigate to={URL.PRODUCTS} />}
```

File: src/shared/index.tsx [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export const URL = {
+  getProducts: "/products",
```
Comment (amitmarko) [inline src/shared/index.tsx]:
```suggestion
  PRODUCTS: "/products",
```

**Review by amitmarko — APPROVED** [review]
LGTM

**Pull request merged**